### ISSUE TITLE
Casting Char to Int fail signedness checker

### ISSUE THREAD

**Issue by VictorYYW** [typetools/checker-framework#5256]
```java
final class C {
  char c;
  public int foo() {
    int x = 1;
    x = x + (int) c;
    return x;
  }
}
```
gives 
```
error: [return.type.incompatible] incompatible types in return.
    return x;
           ^
  type of expression: @Unsigned int
  method return type: @Signed int
```
However, this is okay.
```java
final class C {
  public int foo() {
    char c = 65535;
    int x = 1;
    x = x + (int) c;
    return x;
  }
}
```
I wonder if it is a false positive.

My current workaround is to trade the error for a warning.
```java
import org.checkerframework.checker.signedness.qual.SignedPositive;

final class C {
  char c;
  public int foo() {
    int x = 1;
    x = x + (@SignedPositive int) c;
    return x;
  }
}
```
results in
```
warning: [cast.unsafe] cast from "@Unsigned char" to "@SignednessGlb int" cannot be statically verified
    x = x + (@SignedPositive int) c;
            ^
```

**Comment by mernst** [comment]
Thanks for your bug report.  I'm sorry you are having trouble.

What version of the Checker Framework are you using?
I was unable to reproduce your problem on the latest version of the Checker Framework from its version control repository.  You can see that https://github.com/typetools/checker-framework/pull/5257 is passing tests.

**Comment by VictorYYW** [comment]
Thanks for looking into it, Michael!

You are right. I am using an old version (3.6.0) of the Checker Framework.

Sorry about the confusion. I will close the issue.
